### PR TITLE
chore: add eslint-plugin-jsonc to lint jsonc files

### PR DESCRIPTION
**Pull request by Nemikolh**
This PR adds `eslint-plugin-jsonc` to make it possible to lint `json(c)` files as well with the same rules.

A few notes regarding linting `*.json(c)` files:
  - You'll probably want to update your `.eslintignore` file to ignore generated json files.
  - `eslint-plugin-jsonc` generate different node types for a couple of item (see full list there: https://github.com/ota-meshi/jsonc-eslint-parser/blob/master/src/parser/ast.ts#L33) but the root node still has the `Program` type which is why our lints still works :tada:
  - If you want to disable a specifc rule you can add this to `overrides` in `.eslintrc`:
```jsonc
{
    "files": ["*.json"],
    "rules": { /* ... */ }
}
```

### CODE REVIEW

**Review by d3lm — APPROVED** [review]
This looks good! 🎉 Would it be possible to add a test for a jsonc file to verify that it works as expected?

**Comment by d3lm** [comment]
For the test we could be inspired by this setup https://github.com/ota-meshi/eslint-plugin-jsonc/blob/master/tests/lib/rules/comma-dangle.ts (as an example).

**Comment by Nemikolh** [comment]
Oh thanks a lot for the pointer ! 🎉 Will add that ASAP.

File: test/rules/comment.spec.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { stripIndent } from 'common-tags';
 import { fromFixture } from 'eslint-etc';
-import rule, { defaultOptions, ruleName } from '../../src/rules/comment';
-import { ruleTester } from '../utils';
+import rule, { defaultOptions, ruleName } from '../../src/rules/comment-syntax';
```
Comment (Nemikolh) [inline test/rules/comment.spec.ts:3]: Turns out the test didn't pass when I ran them locally without this change. :sweat_smile:

Comment (d3lm) [reply test/rules/comment.spec.ts:3]: Oh! good catch, ups.

Comment (d3lm) [reply test/rules/comment.spec.ts:3]: I think I changed the name at some point and prolly forgot to change this here. We should start running tests on CI via GitHub Actions I think. We can do this in another PR.

File: test/rules/comment.spec.ts [hunk 2]
```diff
@@ -215,3 +215,37 @@ ruleTester().run(ruleName, rule, {
...
+         */
+        "lorem": true
+      }
+    `
+  ],
+  invalid: [
+    {
+      code: stripIndent`
+      {
+        // Oopsie I've been a baaad boy.
```
Comment (d3lm) [inline test/rules/comment.spec.ts:241]: 🤣

File: test/rules/lines-around-comment.spec.ts [hunk 3]
```diff
@@ -194,3 +194,71 @@ ruleTester().run(ruleName, rule, {
...
+        {
+          beforeLineComment: true,
+        },
+      ],
+      errors: [
+        'Expected line before comment.'
+      ],
+    },
+  ],
+});
```
Comment (d3lm) [inline test/rules/lines-around-comment.spec.ts]: There is a newline missing at the end of the file here.

File: test/rules/comment.spec.ts [hunk 4]
```diff
@@ -215,3 +215,37 @@ ruleTester().run(ruleName, rule, {
...
+        "despicable": 3
+      }
+      `,
+      errors: [
+        'Line comment cannot start with a capital letter unless the entire word is capitalized.',
+        'Line comment cannot end with a dot.'
+      ],
+    },
+  ],
+});
```
Comment (d3lm) [inline test/rules/comment.spec.ts]: Newline missing at the end of the file.

**Review by d3lm — CHANGES_REQUESTED** [review]
Only small remarks but this looks good! Awesome 💪

**Comment by d3lm** [comment]
I think the newlines can be fixed by adding a `.gitattributes` with 
```
* text=auto eol=lf
```

**Review by d3lm — APPROVED** [review]
Good job on this!

**Pull request merged**